x-pack/test/functional/services/ml/data_visualizer_table.ts

```diff
@@ -565,6 +565,18 @@ export function MachineLearningDataVisualizerTableProvider(
       }
     }
 
+    public async assertLensActionShowChart(fieldType: string, fieldName: string) {
+      await retry.tryForTime(10 * 1000, async () => {
+        await testSubjects.clickWhenNotDisabled(
+          this.rowSelector(fieldName, 'dataVisualizerActionViewInLensButton')
+        );
+        await testSubjects.existOrFail('lnsVisualizationContainer', {
+          timeout: 1000,
+        });
+        await browser.goBack();
```

Comment:
I'm a bit concerned about this line because a) the method name `assertLensActionShowChart` doesn't imply that it's also going back and b) we're not validating that the `browser.goBack()` actually did anything and we're not waiting for the page load which could lead to flaky test execution.
I'd suggest to remove this line from here and have a separate method, that we then explicitly call from the test files.
For example we could have something like this in the `ml.navigation` service:
```typescript
async browserBackTo(backTestSubj: string) {
  await browser.goBack();
  await PageObjects.header.waitUntilLoadingHasFinished();
  await testSubjects.existOrFail(backTestSubj, { timeout: 10 * 1000 });
},
```
that we could call right after the charts check like this:
```typescript
if (lensMetricField) {
  await ml.dataVisualizerTable.assertLensActionShowChart(lensMetricField.fieldName);
  await ml.navigation.browserBackTo('dataVisualizerIndexPage');
}
```